storage/src/main/java/tech/pegasys/teku/storage/server/Database.java

```diff
@@ -247,4 +249,45 @@ UInt64 pruneFinalizedBlocks(
 
   Optional<UInt64> pruneFinalizedStates(
       Optional<UInt64> lastPrunedSlot, UInt64 lastSlotToPruneStateFor, long pruneLimit);
+
+  // Sidecars
+  Optional<UInt64> getFirstCustodyIncompleteSlot();
+
+  Optional<UInt64> getFirstSamplerIncompleteSlot();
+
+  Optional<DataColumnSidecar> getSidecar(DataColumnSlotAndIdentifier identifier);
+
+  Optional<DataColumnSidecar> getNonCanonicalSidecar(DataColumnSlotAndIdentifier identifier);
+
+  @MustBeClosed
+  Stream<DataColumnSlotAndIdentifier> streamDataColumnIdentifiers(
+      UInt64 firstSlot, UInt64 lastSlot);
+
+  @MustBeClosed
```

Comment:
we should only provide the start/end version of this, the single slot is only used in one place other than tests.